cmd/oras/internal/option/remote.go

```diff
@@ -149,14 +149,36 @@ func (opts *Remote) Parse(cmd *cobra.Command) error {
 	if cmd.Flags().Lookup(passwordFromStdinFlag) != nil {
 		passwordAndIdTokenFlags = append(passwordAndIdTokenFlags, passwordFromStdinFlag)
 	}
-	cmd.MarkFlagsMutuallyExclusive(usernameAndIdTokenFlags...)
-	cmd.MarkFlagsMutuallyExclusive(passwordAndIdTokenFlags...)
+	if err := CheckMutuallyExclusiveFlags(cmd, usernameAndIdTokenFlags, passwordAndIdTokenFlags); err != nil {
+		return err
+	}
 	if err := opts.parseCustomHeaders(); err != nil {
 		return err
 	}
 	return opts.readSecret(cmd)
 }
 
+// CheckMutuallyExclusiveFlags checks if any mutually exclusive flags are both
+// set, returns an error when detecting set exclusive flags.
+func CheckMutuallyExclusiveFlags(cmd *cobra.Command, exclusiveFlagSets ...[]string) error {
```

Comment:
Add a comment on how the argument `...[]string` will be used, e.g. it detects the first set of conflict and returns the error without checking all the flag sets.